### ISSUE TITLE
Malware reports against ndt7-client.exe

### ISSUE THREAD

**Issue by mattmathis** [m-lab/ndt7-client-go#47]
hybrid-analysis.com and www.joesandbox.com tagged a windows ndt7 binary as malware.

Given the current problems with ndt 7 on mlab3's at 17 sites, I was wondering if this might be a rogue binary that uses ndt7 as chaff to cover something nefarious.   See:
https://www.joesandbox.com/analysis/208192/0/html 
https://www.hybrid-analysis.com/sample/b8af2d7c80de793120bf95211ecb6956366dbd6d9b8a705f909de0cb9f8ee1d6?environmentId=120

A google search for "ndt7-client.exe" (include the quotes) finds multiple reports similar to the above.

**Comment by mattmathis** [comment]
I got a free account at joesandbox, to investigate if I cam figure out where they collected the binary.

**Comment by robertodauria** [comment]
@mattmathis I'm going through long-standing issues to see if there is any need to update them.

I read the reports above and pretty much everything seems just a false positive to me, or a demonstration that these tools do not work very well with Go binaries;

- Package names are detected as "potential email addresses":
```
Pattern match: "tcp-info@v1.3.0"
Pattern match: "websocket@v1.4.1"
Pattern match: "go@v1.2.0" 
```

- The `wine_get_version` syscall isn't a malicious indicator -- the Go runtime actively probes to see if it's running under Wine (https://go.dev/src/runtime/os_windows.go - search for `wine_get_version`)

- The monitored registry key all have to do with TLS certificates. I haven't looked at Go's crypto library in detail, but since we're using TLS in the client it seems natural that the library would read the valid CAs, etc. from the registry. In a Linux environment it probably just reads some files.

- Some things that are definitely not IP addresses are detected as IP addresses
```
Heuristic match: "crypto/tls.(*clientHelloMsg).marshal.func1.4.1.1.1"
Heuristic match: "crypto/tls.(*clientHelloMsg).marshal.func1.4.9.1.1"
Heuristic match: "crypto/tls.(*clientHelloMsg).marshal.func1.4.12.1.1"
``` 

- Traffic over ports 80/443 without a HTTP header is expected, since we're using WebSocket
- The contacted domains look as expected from an ndt7 client:
```
details
    "locate.measurementlab.net"
    "ndt-iupui-mlab2-ord03.measurement-lab.org" 
```
- The IP addresses look OK, too:
```
details
    "172.217.15.115:443" (this is a google cloud ip address, likely locate.measurementlab.net when accessed from the US)
    "66.198.24.88:443" (this is mlab2-ord03)
```

- All the URLs in memory are either Go package names, or an URL we know about (e.g. Locate)
- All the VirusTotal detections are "generic" ones, likely based on the same criteria that joesandbox and hybrid-analysis are using. joesandbox in particular gives it a very high "malicious score" just because it is detected by some AVs on VirusTotal.

Is there anything I'm missing?

In short... This is, IMHO, very likely to be a legitimate ndt7-client binary. A plausible test to confirm could be to build the version of ndt7-client that was available at the time with the Go version available at the time for Windows, submit it for analysis to the same services and compare.

If we can figure out if this still happens with recent versions of Go and ndt7-client on Windows, we may also want to add a notice into this repository's README. Please let me know if you'd like to experiment more. I would be inclined to close this issue and re-open only if there are other reports or updates.

**Comment by mattmathis** [comment]
Yes, likely a false positive.  Everything the scanners flagged as suspicious is an expected property of the application.
